File: restapi/app/models/match.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from app import db
from app.models.base import BaseModel

class Match(BaseModel):
	__tablename__ = 'match'
	__json_public__ = ['id', 'homeTeamName', 'homeTeamCode', 'homeTeamFlag', 'awayTeamName', 'awayTeamCode', 'awayTeamFlag', 'date', 'reportTime', 'disputeTime', 'outcomes', 'homeScore', 'awayScore', 'name', 'public', 'market_fee', 'grant_permission', 'source_id', 'category_id', 'creator_wallet_address', 'image_url']
	__json_modifiers__ = {
        'outcomes': lambda outcomes, _: [outcome.to_json() for outcome in outcomes]
    }
	homeTeamName = db.Column(db.String(255))
	homeTeamCode = db.Column(db.String(10))
	homeTeamFlag = db.Column(db.String(512))
	awayTeamName = db.Column(db.String(255))
	awayTeamCode = db.Column(db.String(10))
	awayTeamFlag = db.Column(db.String(512))
	name = db.Column(db.String(512))
	homeScore = db.Column(db.Integer)
	awayScore = db.Column(db.Integer)
	market_fee = db.Column(db.Integer,
							server_default=str(1),
	                      	default=1)
	date = db.Column(db.BigInteger) #closingTime

	reportTime = db.Column(db.BigInteger)
	disputeTime = db.Column(db.BigInteger)
	index = db.Column(db.Integer,
							server_default=str(1),
	                      	default=1)
	public = db.Column(db.Integer,
						server_default=str(1),
						default=0)
	# this field help admin set outcome for creator
	grant_permission = db.Column(db.Integer,
						server_default=str(0),
						default=0)
	creator_wallet_address = db.Column(db.String(255))
	image_url = db.Column(db.String(512))
	category_id = db.Column('category_id', db.ForeignKey('category.id'))
	source_id = db.Column('source_id', db.ForeignKey('source.id'))
	outcomes = db.relationship('Outcome', backref='match', primaryjoin="Match.id == Outcome.match_id", lazy='dynamic')

	@classmethod
	def find_match_by_id(cls, match_id):
		return Match.query.filter_by(id=match_id).first()

	def __repr__(self):
		return '<match {}, {}, {}>'.format(self.id, self.homeTeamName, self.awayTeamName)

File: event/models/match.go
package models

// Source : model
type Source struct {
	ID   int
	Name string `gorm:"column:name;" json:"name"`
	URL  string `gorm:"column:url;" json:"url"`
}

type Category struct {
	ID   int
	Name string `gorm:"column:name;" json:"name"`
}

// TableName : source
func (s Source) TableName() string {
	return "source"
}

// Match : model
type Match struct {
	ID          int
	CloseTime   int       `gorm:"column:date;" json:"date"`
	ReportTime  int       `gorm:"column:reportTime;" json:"reportTime"`
	DisputeTime int       `gorm:"column:disputeTime;" json:"disputeTime"`
	Name        string    `gorm:"column:name;" json:"name"`
	Source      Source    `gorm:"foreignkey:SourceID"`
	SourceID    int       `gorm:"column:source_id;" json:"source_id"`
	Category    Category  `gorm:"foreignkey:CategoryID"`
	CategoryID  int       `gorm:"column:category_id;" json:"category_id"`
	OutcomeName string    `gorm:"column:name;" json:"outcome_name"`
	EventName   string    `gorm:"column:name;" json:"event_name"`
	Outcomes    []Outcome `gorm:"foreignkey:MatchID"`
}

// TableName : match
func (m Match) TableName() string {
	return "match"
}
